frontend/src/visualization/filtering/FilterLineProperty.tsx

```diff
@@ -7,23 +7,59 @@ import {
   MenuItem,
   Select,
 } from '@material-ui/core';
-import React from 'react';
-import FilterPropertyModel from '../../FilterPropertyModel';
+import React, { useEffect } from 'react';
+import useService from '../../dependency-injection/useService';
+import FilterStateStore from '../../stores/filterState/FilterStateStore';
+import { FilterModelEntry } from '../../shared/filter';
 
 const useStyles = makeStyles(() =>
   createStyles({
     select: {
-      width: 200,
+      width: 500,
     },
   })
 );
 
-const EntityPropertySelect = (props: {
-  property: FilterPropertyModel;
-  setProperty: React.Dispatch<React.SetStateAction<FilterPropertyModel>>;
+const FilterLineProperty = (props: {
+  filterModelEntry: FilterModelEntry;
+  filterLineType: string;
+  entity: 'node' | 'edge';
+  applyFilter: boolean;
+  setApplyFilter: React.Dispatch<React.SetStateAction<boolean>>;
 }): JSX.Element => {
   const classes = useStyles();
-  const { property, setProperty } = props;
+  const {
+    filterModelEntry,
+    filterLineType,
+    entity,
+    applyFilter,
+    setApplyFilter,
+  } = props;
+
+  const filterStateStore = useService<FilterStateStore>(FilterStateStore);
+
+  const [selectedValues, setSelectedValues] = React.useState<string[]>(
+    filterStateStore.getPropertyStateValues(
+      filterLineType,
+      filterModelEntry.key,
+      entity
+    ) ?? []
+  );
+
+  // update filterStateStore here because selectedValues will first be updated in the next render
```

Comment:
why don't you just call `filterStateStore.addFilterPropertyState(...)` in `handleChange`? The current implementation can result in unwanted side effects if multiple `FilterLineProperty`s exist

Reply:
See the comment I wrote above, a `useState` variable is first updated in the next render. See: https://stackoverflow.com/questions/54069253/usestate-set-method-not-reflecting-change-immediately

Reply:
but can't `applyFilter` only happen after the apply filter was clicked and thus after the select is closed?